### PR TITLE
Use assign for deserialize instead of temporary std::string

### PR DESCRIPTION
**Pull request by SamuelH91**
## Description

At deserialize of fixed_string (cdr) or string (fastcdr) temporary std::string was used which may allocate memory dynamically, which is not desirable or needed especially if using fixed_strings.

@Mergifyio backport 1.0.x

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_ Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- [x] Changes are API compatible. 
- _N/A_ New feature has been added to the `versions.md` file (if applicable).
- [x] Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: CI pass and failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/Cdr.cpp [hunk 1]
```diff
@@ -1658,7 +1658,7 @@ Cdr& Cdr::deserialize(
 const char* Cdr::read_string(
         uint32_t& length)
 {
-    const char* ret_value = "";
+    const char* ret_value = nullptr;
```
[the remark this answers is not in the document]
Comment (SamuelH91) [reply src/cpp/Cdr.cpp]: Reverted

File: src/cpp/FastCdr.cpp [hunk 2]
```diff
@@ -452,7 +452,7 @@ FastCdr& FastCdr::deserialize(
 const char* FastCdr::read_string(
         uint32_t& length)
 {
-    const char* ret_value = "";
+    const char* ret_value = nullptr;
```
[the remark this answers is not in the document]
Comment (SamuelH91) [reply src/cpp/FastCdr.cpp]: Reverted

**Comment by MiguelCompany** [comment]
@mergifyio rebase

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Comment by MiguelCompany** [comment]
@SamuelH91 Thank you for your contribution and patience!

**Comment by MiguelCompany** [comment]
@Mergifyio backport 1.0.x

**Pull request merged**